src/nvim/optionstr.c

```diff
@@ -1594,6 +1594,8 @@ static void do_spelllang_source(win_T *win)
   if (p > q) {
     vim_snprintf(fname, sizeof(fname), "spell/%.*s.vim", (int)(p - q), q);
     source_runtime(fname, DIP_ALL);
+    vim_snprintf(fname, sizeof(fname), "spell/%.*s.lua", (int)(p - q), q);
```

Comment:
Didn't zeer do a related change recently where combining these lines was significantly faster?

Reply:
Sounds like something @zeertzjq would do :D 

Where would I look for ~~stealing~~ learning about that?

Reply:
I notice that the pattern for compilers and for colorschemes is to only load the Lua file if no Vim file is found; is this something we want for these spell files as well?

(I have zero idea what these files are used for in practice...)

Reply:
@clason See https://github.com/neovim/neovim/pull/21942.

Reply:
Sorry, that pointer arithmetic is doing my head in. If somebody knows how to do that, feel free to tell me or take over the PR.

Reply:
> I notice that the pattern for compilers and for colorschemes is to only load the Lua file if no Vim file is found; is this something we want for these spell files as well?

If that is the case, then these patterns cannot be combined.

Reply:
I see no reason to give vimscript a higher priority. We should combine these if we can.

Reply:
No pointer arithmetic is needed for this PR anyway.

Reply:
> I see no reason to give vimscript a higher priority. We should combine these if we can.

We give vimscript a higher priority _everywhere_ (we always source vim files first, then Lua files)?

Reply:
Which means we need to search everywhere twice. For niche things like this which very few people use, this makes it an overall net-loss. Since majority of users will be paying the searching cost.

What's the reason for prioritizing vimscript? Seems like an overly conservative decision we should revisit.

Reply:
Consistency, plain and simple. The source order is documented.

Reply:
(If that's what you are asking about. If it's about having Lua only as a fallback to Vim, I have no idea. Again, I don't know what these files are actually useful for.)

Reply:
Is consistency worth the performance cost for this specifically?

Reply:
> I'm referring to this `(int)(p-q)` business. No idea how to do that in a combined source.

No change to that part is needed. You just need to change `.vim` to `.\\(vim\\|lua\\)`.

Reply:
Do you actually see a significant performance issue here? It's only sourcing a single file or two. (This is a very different situation than https://github.com/neovim/neovim/pull/21942.)

Reply:
Isn't it searching over `rtp`? If so then I think that's significant enough considering a large majority of users don't use this. Especially since there's an alternative of combining them.